### PR TITLE
Docs for Flow 0.12

### PR DESCRIPTION
**Pull request by mmrj**
includes changelog updates, but i haven't tagged yet -- converterargs needs to be merged to master also before we tag

### CODE REVIEW

File: src/converters/bool-conditional-converters.js [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+ * For a boolean conditional converter, the original format is your model variable, and the resulting "format" is a boolean value, or another value of your choosing.
+ *
+ * There are two ways to specify conversion or formatting for the display output of a particular model variable:
+ *
+ * * Use the `|` (pipe) character within the value of any `data-f-` attribute.
+ * * Add the attribute `data-f-convert` to any element that also has the `data-f-bind` or `data-f-foreach`.
+ *
+ * For example:
+ *
+ *      <!-- will show "true" or "false" -->
```
Comment (narenranjit) [inline src/converters/bool-conditional-converters.js]: think it's worth calling out the "truthiness' with examples

```
 <span data-f-bind="sampleVar | toBool"></span> <!-- true if 'sampleVar' is truthy (1,'some string', [] etc), false if 'sampleVar' is falsy (0, false, '') -->
```

File: src/converters/bool-conditional-converters.js [hunk 2]
```diff
@@ -14,17 +34,54 @@ function parseArgs(trueVal, falseVal, input, matchString) {
...
+     *
+     *      <div>
+     *          <span data-f-bind="sampleVar | ifTrue('yes! please move forward', 'not ready to proceed')"></span> 
+     *          <span data-f-bind="sampleVar | ifTrue('yes! please move forward')"></span>
+     *      </div>
+     *
+     * @param {Number} trueVal The value to display if the input is true.
+     * @param {Number} falseVal (Optional) The value to display if the input is false. If not included, returns `false`.
+     * @param {Any} input (Optional) The input to test. If not included, the output of the previous argument is used.
+     * @return {Any} If input is true, returns trueVal. If input is false, returns falseVal if provided, or `false`.
```
Comment (narenranjit) [inline src/converters/bool-conditional-converters.js]: This was supposed to return the original value if not true and no falseval specified; is that not happening?

File: src/converters/bool-conditional-converters.js [hunk 3]
```diff
@@ -14,17 +34,54 @@ function parseArgs(trueVal, falseVal, input, matchString) {
...
+     *
+     *      <div>
+     *          <span data-f-bind="sampleVar | ifFalse('not ready to proceed', 'actually this is still true')"></span> 
+     *          <span data-f-bind="sampleVar | ifFalse('not ready to proceed')"></span>
+     *      </div>
+     *
+     * @param {Number} trueVal The value to display if the input is false.
+     * @param {Number} falseVal (Optional) The value to display if the input is true. If not included, returns `false`.
+     * @param {Any} input (Optional) The input to test. If not included, the output of the previous argument is used.
+     * @return {Any} If input is false, returns trueVal. If input is true, returns falseVal if provided, or `false`.
```
Comment (narenranjit) [inline src/converters/bool-conditional-converters.js]: see prev comment

File: src/converters/number-compare-converters.js [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+ * There are two ways to specify conversion or formatting for the display output of a particular model variable:
+ *
+ * * Use the `|` (pipe) character within the value of any `data-f-` attribute.
+ * * Add the attribute `data-f-convert` to any element that also has the `data-f-bind` or `data-f-foreach`.
+ *
+ * For example:
+ *
+ *      <!-- displays "true" or the number of widgets -->
+ *      <span data-f-value="widgets | greaterThan(50)"></span>
+ *      <span data-f-value="widgets" data-f-convert="greaterThan(50)"></span>
```
Comment (narenranjit) [inline src/converters/number-compare-converters.js]: don't think this is useful as a `data-f-convert` (unless you want to compare every bound element within this as to 50 which seems a stretch as a use-case), so we can remove that.

Also should be bind, not value

File: src/converters/number-compare-converters.js [hunk 5]
```diff
@@ -14,23 +38,118 @@ function parseArgs(limit, trueVal, falseVal, valueToCompare, matchString) {
...
+     *          <span data-f-bind="widgets | greaterThan(50)"></span> 
+     *
+     *          <!-- displays custom text -->
+     *          <span data-f-bind="widgets | greaterThan(50, 'Congratulations!', 'Better luck next year')"></span>
+     *      </div>
+     *
+     * @param {Number} limit The reference value to compare the model variable against.
+     * @param {Number} trueVal (Optional) The format (value) to display if the model variable is greater than `limit`. If not included, the display is `true`.
+     * @param {Number} falseVal (Optional) The format (value) to display if the model variable is less than or equal to `limit`. If not included, the display is the value of the model variable.
+     * @return {Any} If the model variable is greater than `limit`, returns trueVal or `true`, otherwise returns falseVal or `false`.
```
Comment (narenranjit) [inline src/converters/number-compare-converters.js]: returns falseval if provided or echoes the input

File: src/converters/number-compare-converters.js [hunk 6]
```diff
@@ -14,23 +38,118 @@ function parseArgs(limit, trueVal, falseVal, valueToCompare, matchString) {
...
+     *          <span data-f-bind="widgets | greaterThan(50)"></span> 
+     *
+     *          <!-- displays custom text -->
+     *          <span data-f-bind="widgets | greaterThan(50, 'Congratulations!', 'Better luck next year')"></span>
+     *      </div>
+     *
+     * @param {Number} limit The reference value to compare the model variable against.
+     * @param {Number} trueVal (Optional) The format (value) to display if the model variable is greater than or equal to `limit`. If not included, the display is `true`.
+     * @param {Number} falseVal (Optional) The format (value) to display if the model variable is less than `limit`. If not included, the display is the value of the model variable.
+     * @return {Any} If the model variable is greater than or equal to `limit`, returns trueVal or `true`, otherwise returns falseVal or the value of the model variable.
```
Comment (narenranjit) [inline src/converters/number-compare-converters.js]: returns falseval if provided or echoes the input (also for all other converters here)

File: src/converters/number-compare-converters.js [hunk 7]
```diff
@@ -14,23 +38,118 @@ function parseArgs(limit, trueVal, falseVal, valueToCompare, matchString) {
     }
 }
 module.exports = {
+    /**
+     * Convert the model variable to true, or other values passed to the converter, based on whether the model variable is greater than the limit.
```
Comment (narenranjit) [inline src/converters/number-compare-converters.js]: "converts the model variable" sounds a bit like it's changing the value in the model, maybe "converts the model variable output" would be more accurate?

File: src/converters/number-compare-converters.js [hunk 8]
```diff
@@ -14,23 +38,118 @@ function parseArgs(limit, trueVal, falseVal, valueToCompare, matchString) {
...
+     *      <div>
+     *          <!-- displays true or the number of widgets -->
+     *          <span data-f-bind="widgets | equalsNumber(50)"></span> 
+     *
+     *          <!-- display custom text -->
+     *          <span data-f-bind="widgets | equalsNumber(50, 'Met the goal exactly!', 'Not an exact match')"></span>
+     *      </div>
+     *
+     * @param {Number} limit The reference value to compare the model variable against.
+     * @param {Number} trueVal (Optional) The format (value) to display if the model variable is greater than `limit`. If not included, the display is `true`.
```
Comment (narenranjit) [inline src/converters/number-compare-converters.js]: copy-paste error, should say equals, not greater than

File: src/dom/attributes/toggles/hide-if-attr.js [hunk 9]
```diff
@@ -1,3 +1,22 @@
...
+ * ## Display Elements Conditionally (hideif)
+ *
+ * The `data-f-hideif` attribute allows you to hide DOM elements based on either the value of the model variable (true or false), or the value of a [comparison (Number Comparison Converter)](../../../../converters/number-compare-converter/) using a model variable.
+ *
+ * **Examples:**
+ *
+ *      <!-- model variable already has a boolean value -->
+ *      <div data-f-hideif="sampleBooleanModelVariable">Remains hidden if the model variable is true</div>
+ *
+ *      <!-- chain with the greaterThan converter to produce a boolean value,
```
Comment (narenranjit) [inline src/dom/attributes/toggles/hide-if-attr.js]: The equivalent of the above would be piping to `toBool`; but this example is useful as-is so we should keep all 3

File: documentation/attributes-overview/index.html.md [hunk 10]
```diff
@@ -21,23 +21,18 @@ Once you set `data-f-bind`, Flow.js figures out the appropriate action to take b
...
+| Display and automatically update a variable in the interface?                    | `data-f-bind`                          | See [Default Bi-directional Binding](../generated/dom/attributes/binds/default-bind-attr/). |
+| Display and automatically update array and object variables?                     | `data-f-foreach`, and optionally, [templates](../#templates)  | See [Display Array and Object Variables](../generated/dom/attributes/foreach/default-foreach-attr/).   |
+| Display a DOM element based on whether a variable is true or false?              | `data-f-showif`, `data-f-hideif`       | See [Display Elements Conditionally (showif)](../generated/dom/attributes/toggles/show-if/) and [Display Elements Conditionally (hideif)](../generated/dom/attributes/toggles/hide-if/). |
+| Display the value of the variable at every time step (not just the current one)? | `data-f-repeat`   | See [Display Array Variables](../generated/dom/attributes/repeat-attr/).               |
+| Display a variable only (read-only binding)?                                     | `data-f-value`, for example            | See [read only binding](../generated/dom/attributes/default-attr/).  |
```
Comment (mmrj) [inline documentation/attributes-overview/index.html.md]: followup question: do you think http://qa.forio.com/epicenter/docs/public/data_binding_flow_js/generated/dom/attributes/default-attr/ is a page worth having then? maybe we shouldn't even include this?

Comment (mmrj) [inline documentation/attributes-overview/index.html.md]: (or http://qa.forio.com/epicenter/docs/public/data_binding_flow_js/generated/dom/attributes/default-attr/ if you want to be on a server that has this pr -- the default-attr page is the same on both)

Comment (mmrj) [inline documentation/attributes-overview/index.html.md]: planning to keep that page and reference only under `data-f-<attribute>` row, but let me know if you disagree.

File: documentation/graphing-overview/index.html.md [hunk 11]
```diff
@@ -163,8 +163,8 @@ Here's the complete sample code:
 		<script src="//forio.com/tools/contour/contour.js"></script>
 
 		<!-- for Flow.js -->
-		<script src="//ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
-		<script src="//forio.com/tools/js-libs/2.1.0/epicenter.min.js"></script>
+		<script src="//ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js"></script>
+		<script src="//forio.com/tools/js-libs/2.2.0/epicenter.min.js"></script>
```
Comment (mmrj) [inline documentation/graphing-overview/index.html.md]: not currently tracking patch versions...

File: src/converters/bool-conditional-converters.js (hunk 2 above)
Comment (mmrj) [inline src/converters/bool-conditional-converters.js]: you're right. after my confusion on truthiness i was only using boolean model variables in testing.  but `sampleZeroVar = 0` and `<span data-f-bind="sampleZeroVar | ifTrue('truestring')"></span>` displays `0`

**Comment by mmrj** [comment]
@narenranjit i think this is all set for re-review.

File: documentation/attributes-overview/index.html.md (hunk 10 above)
Comment (narenranjit) [inline documentation/attributes-overview/index.html.md]: I think calling that if we call that *readonly mode for input elements* (kind of an oxymoron) it's fine -- we could also add information about the readonly flag passed to the channels here. The usecase here is not-too-uncommon (i.e. maybe you toggle a demo-mode which is output only) so might be worth keeping

File: documentation/graphing-overview/index.html.md (hunk 11 above)
Comment (narenranjit) [inline documentation/graphing-overview/index.html.md]: Tangentially, make a ticket if it's helpful for me to change the build process to centralize all the version numbers in a single json file which applies to all the doc files.

File: documentation/converter-overview/index.html.md [hunk 12]
```diff
@@ -7,21 +7,31 @@ isPage: true
...
-Converters allow you to convert data -- for example, values of variables in your model that you display in your project's user interface -- from one form to another.
+Converters allow you to change how data is displayed. They let you display the value of any model variable in a different format than it is stored in the model -- converting the output value from one format to another.
 
 There are two ways to specify conversion or formatting for the display output of a particular model variable:
 
-* Add the attribute `data-f-convert` to any element that also has the `data-f-bind` or `data-f-foreach`.
-* Use the `|` (pipe) character within the value of any `data-f-` attribute (not just `data-f-bind` or `data-f-foreach`).
+* Use the `|` (pipe) character within the value of any `data-f-` attribute. Converters are chainable, so you can apply several in a row to a particular variable:
+		
+		<span data-f-bind="widgets | last | #,###"
```
Comment (narenranjit) [inline documentation/converter-overview/index.html.md]: need to close tag (nitpick, but people do copy-paste :) )

**Pull request merged**